backend/src/rules/victory_conditions.clj

```diff
@@ -18,40 +18,33 @@
       0
       row-data)))
 
-(defn get-points
-  "Tells us how many points each player has on each row"
-  [game-state]
-  (loop [row 0
-         points []]
-    (if (> row 4)
-      points
+(defn get-won-rows
```

Comment:
two things: 

1. I would define `(let [opponent (mod (inc player) 2)]` at the beginning of the function.
2. I think this would look better as a `reduce` than a `loop`

Reply:
some clue regarding how to do that? :)

Reply:
This quick solution requires redoing `points-in-row`:

```clojure
(defn points-in-row
  "Tells us how many points a certain row contains for a certain player"
  [game-state row]
  (reduce
    #(if (= (:owner %2) player)
       (+ %1 (:power %2))
       %1)
    0
    row))

(defn winning-player?
  [row player]
  (let [opponent (mod (inc player))]
    (> (points-in-row row player)
       (points-in-row row opponent))))

(defn get-won-rows
 "Tells us how many rows a player is winning"
 [game-state player]
 (reduce
   #(if (winning-player? %2 player) (inc %1) %1)
   0
   (get-in game-state [:rows])))
```

By the way, I didn't find a way to quickly chech if points-in-row is used elsewhere. There's one thing we should **really** start doing, and it's marking private functions as private by declaring them with `(defn-` which dissallows using them outside their namespace. That way we know which functions are only extracted logic and don't need to be tested.

Reply:
Also, if preferred, `(defn ^:private points-in-row` is longer, but clearer, and for some reason, the only way to do private `(def`s ( there is no `(def-`)